crates/nu-command/src/filesystem/ucp.rs

```diff
@@ -166,6 +166,17 @@ impl Command for UCp {
                         if f.is_empty() {
                             return Err(ShellError::FileNotFound(p.span));
                         }
+                        let any_source_is_dir = f.iter().any(|f| matches!(f, f if f.is_dir()));
+                        if any_source_is_dir && !recursive {
+                            return Err(ShellError::GenericError(
+                                "Directories must be copied using \"--recursive\"".into(),
+                                "resolves to a directory (not copied)".into(),
+                                Some(p.span),
+                                None,
```

Comment:
may i propose
```suggestion
                                "could_not_copy_directory".into(),
                                "resolves to a directory (not copied)".into(),
                                Some(p.span),
                                Some("Directories must be copied using \"--recursive\"".into()),
```

to get an error like
```
Error:   × could_not_copy_directory
   ╭─[entry #2:1:1]
 1 │ ucp foo/ bar
   ·     ──┬─
   ·       ╰── resolves to a directory (not copied)
   ╰────
  help: Directories must be copied using "--recursive"
```
? :innocent:

Reply:
Yes, I like your error better. Changed :)

Reply:
amazing thanks :pray: :star_struck: